server.go

```diff
@@ -1500,6 +1506,13 @@ func (s *Server) ReadSeries(database, retentionPolicy, name string, tags map[str
 // Returns a resultset for each statement in the query.
 // Stops on first execution error that occurs.
 func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Results {
+	// Authorize user to execute the query.
+	if err := Authorize(user, q, database); err != nil {
```

Comment:
Auth is not a failure. A failure is a `500`. Auth is a validation that throws `401` if it didn't pass. I think it's entirely consistent. Errors in the 400 range should be evaluated before actual query execution actually begins.